doc/tutorials/specification-language.md

```diff
@@ -0,0 +1,126 @@
+configuration settings. Setting metadata should be done in the spec namespace. A metadata is just additional data describing the configuration setting.
+One way to set metadata is the command line tool [kdb meta-set](../help/kdb-meta-set.md) which we have used in the upper example.
+For the [validation plugin](../../src/plugins/validation/README.md) which supports regular expression checks there exists even its own
+command line tool [kdb vset](../help/kdb-vset.md).
+
+If you are writing validation for dozens of configuration settings, the command line might not be the most convenient way. You can also
+write specifications in any file format of your desire. Examples can be seen [here](../../examples/spec). We recommend the INI
+file format as it natively supports metadata in an easy and understandable way. So the upper example
+could be rewritten in an INI file in the following way:
+
+```ini
+[]
+mountpoint = test-specification.ini
+infos/plugins = type
+
+[bool]
+type = boolean
+```
+
+To integrate a configuration into the key database, we mount it by using the line `mountpoint = test-specification.ini`
+in the configuration specification at the top. This is a mandatory information for the path under which the configuration is mounted
+(also called the `parent key`).
+This will be mounted into the spec namespace with the following command:
+
+```shell script
+kdb mount </absolute/path/to/ini/file.ini> spec/tests ni
+kdb spec-mount /tests
+```
+
+The `spec-mount` command tells Elektra to load all relevant plugins. It is smart enough to automatically detect the needed plugins
+by looking at all metadata. Sine `type` for is given under the `[bool]` key, Elektra will load the `type` plugin.
+You might need to remove some metadata to not load certain plugins that cause the exceeding of the maximum
+plugin number.
+
+What remains to know are the metadata names and meanings. We have used the `type` plugin which provides the metadata `type` metadata along with
+many other metadata such as `check/enum`. You can read all available metadata in the plugin READMEs which all have to have a
+so called `contract` at the beginning of each file. This contract also contains the `infos/metadata` data which lists all available
+metadatas. We recommend to look through those plugins READMEs to know how they are intended to use.
+
+As another example you may take the [range](/src/plugins/range/README.md) plugin. The `infos/metadata` show that
+`check/range` is provided by the plugin:
+
+```shell script
+kdb meta-set /tests/abc 'check/range' '1-10'
+kdb mount test.conf /tests dump range
+kdb set /tests/abc 12
+# Using name user/tests/abc
+# Sorry, module range issued the error C03200:
+# Validation Semantic: Value '12' of key 'user/tests/abc' not within range 1-10
+```
+
+You can also find all available metadata in the [METADATA.ini](/doc/METADATA.ini) file. It shows which
```

Comment:
"also"? Where else? Also mount METADATA.ini and make a shell recorder example with ipaddr.

Reply:
> "also"? Where else?

In the README.md files of every plugin

> Also mount METADATA.ini and make a shell recorder example with ipaddr.

I do not understand exactly what this is supposed to achieve.

Reply:
> I do not understand exactly what this is supposed to achieve.

To understand that meta-meta-data can also be queried within Elektra. Furthermore, the shell recorder tests will wake us up if one day the semantics of check/ipaddr are changed.